dart/test/hub_test.dart

```diff
@@ -396,6 +396,83 @@ void main() {
     });
   });
 
+  group('Hub scope callback', () {
+    var hub = Hub(SentryOptions(dsn: fakeDsn));
+    var client = MockSentryClient();
+
+    SentryLevel? loggedLevel;
+    Object? loggedException;
+
+    void mockLogger(
+      SentryLevel level,
+      String message, {
+      String? logger,
+      Object? exception,
+      StackTrace? stackTrace,
+    }) {
+      loggedLevel = level;
+      loggedException = exception;
+    }
+
+    setUp(() {
+      final options = SentryOptions(dsn: fakeDsn)
+        ..debug = true
+        ..logger = mockLogger; // Enable logging in DiagnosticsLogger
+
+      hub = Hub(options);
+      client = MockSentryClient();
+      hub.bindClient(client);
```

Comment:
We can use the `Fixture` pattern instead of declaring everything under the group and setUp